hack/validate-prow-job-semantics.py

```diff
@@ -236,8 +236,8 @@ def validate_pod_name(path, data):
                 if job["agent"] != "kubernetes":
                     continue
 
-                if job["spec"]["containers"][0]["name"] != "":
-                    logger.error("%s: ci-operator job %s should not set a pod name", path, job["name"])
+                if len(job["spec"]["containers"]) == 1 and job["spec"]["containers"][0]["name"] != "":
```

Comment:
How many tests (with more than one container) are we talking about?
If it is just one or two, maybe the exceptional cases are on the job name, instead of #containers?

Reply:
Hello,
`validate-prow-job-semantics` is the only tests that fails when a name is defined on the first container. When no name is defined on the first container, and there is more than one container in the job, it fails `prow-config-semantics` `deprecate-templates` and `config`. In addition, even in upstream prow, jobs with sidecars require names as shown by the result of the `config` test: `error loading prow config: invalid presubmit job oc-mirror-e2e-unit: [container does not have name. all containers must have names when defining multiple containers, container named \"\" is defined more than once` as referenced at https://github.com/kubernetes/test-infra/blob/master/prow/config/config.go#L2504
You can see an example on a draft PR at https://github.com/openshift/release/pull/23364

Reply:
What i meant is to wave the condition on job name `oc-mirror-e2e-unit`, instead of #containers?

Reply:
Instead of fixing this contradiction in the tests that could affect other users making sidecars as well? There also may be times when we write post-submits or periodics and we would be right back here. So I would feel better if this was future-proofed instead of having to add a list of jobs to an exclusive exclusion list every time. Though I could write something for that and put it in the PR if really need be.

Reply:
I think this is fine. It gives a safe escape hatch for power users, never gets in a way normally.